amplify.yml

```
@@ -0,0 +1,18 @@
+version: 1
+frontend:
+    phases:
+        preBuild:
+            commands:
+                - 'npm ci --cache .npm --prefer-offline'
+        build:
+            commands:
+              - 'if [[ "$AWS_BRANCH" == "main" ]]; then npm run build:staging; fi'
+              - 'if [[ "$AWS_BRANCH" == release* ]]; then npm run build:production; fi'
+              - 'if [[ "$AWS_BRANCH" != "main" && "$AWS_BRANCH" != release* ]]; then npm run build; fi'
```

By default, it should build for staging and only on `release*` branches for production:

```suggestion
              - '[[ "$AWS_BRANCH" == release* ]] && npm run build || npm run build:stage'
```